OrbitCore/SymbolsClient.cpp

```diff
@@ -84,11 +88,32 @@ void SymbolsClient::HandleResponse(const Message& message, uint64_t id) {
   core_app_->OnRemoteModuleDebugInfo(infos);
 
   // Finalize transaction.
-  id_sessions_mutex_.Lock();
+  mutex_.Lock();
   std::shared_ptr<Session> session = id_sessions_[id];
   id_sessions_.erase(id);
-  id_sessions_mutex_.Unlock();
+
+  std::vector<std::string> not_found_module_names =
+      std::move(id_not_found_module_names_[id]);
+  id_not_found_module_names_.erase(id);
+  mutex_.Unlock();
+
   if (session != nullptr) {
     core_app_->ApplySession(*session);
   }
+
+  // Show an error MessageBox for modules not found locally nor remotely.
+  for (const ModuleDebugInfo& module_info : infos) {
+    if (!module_info.m_Functions.empty()) {
+      not_found_module_names.erase(std::find(not_found_module_names.begin(),
+                                             not_found_module_names.end(),
+                                             module_info.m_Name));
+    }
+  }
+  if (!not_found_module_names.empty()) {
+    std::string text = "Could not load symbols for modules:";
+    for (const std::string& module_name : not_found_module_names) {
+      text += "\n  " + module_name;
+    }
+    core_app_->SendErrorToUi("Error loading symbols", text);
```

Comment:
Ah, ok... here it is. Should the SymbolsClient be moved to OrbitGl? in which case it could use a callback or OrbitApp global instead.

Reply:
Well, from the name of the module, no, but apart from that, the question would be whether all the `...Client` classes should be in OrbitGl. I don't know, should they?
But I can see what you mean that the `SendErrorToUi` logic doesn't really belong in `OrbitCore`. I could also just remove `SendErrorToUi` altogether and put that logic somewhere else (although, where?) as the parsing logic is actually in `OrbitMainWindow::OnReceiveMessage` so in `OrbitQt`.
More thoughts?

Reply:
Or indeed we move the `...Client` classes to `OrbitGl` and the `...Service` classes to `OrbitService`, the latter seems to be quite fitting.

Reply:
So.. yes. If you want to do something to UI you probably need something from OrbitApp class which is in OrbitGl, so the code that depends on it should probably be in the Gl or a Module that depends on it. Core does not depend on OrbitGl neither should it. So it seems this particular class belongs to OrbitGl or OrbitQt... we could also introduce new module OrbitClient and move some things from OrbitGl to OrbitClient.

Reply:
Ok, so I added commits to move the `...Client` classes to `OrbitGl` and the `...Service` classes to `OrbitService`. Let's delay `OrbitClient` for now...